weave/trace_server/execution_runner/cross_process_trace_server.py

```diff
@@ -0,0 +1,540 @@
+"""
+Cross-process trace server implementation.
+
+This module provides a way to run any TraceServerInterface implementation in a
+separate thread, providing isolation for concurrent operations.
+
+The main functions are:
+- generate_child_process_trace_server_args:
+  Called in parent process to start a worker thread and get args
+- build_child_process_trace_server: Called to build a client that communicates with the worker
+"""
+
+import contextvars
+import logging
+import multiprocessing
+import queue
+import threading
+from collections.abc import Iterator
+from typing import Any, Optional, TypedDict
+
+from weave.trace_server.secret_fetcher_context import _secret_fetcher_context
+from weave.trace_server.trace_server_interface import *
+
+# Configure logging
+logger = logging.getLogger(__name__)
+
+# List of context variables to propagate to worker threads
+# Add any context variables here that need to be available in the worker thread
+# Example: If you have a custom context var, add it to this list:
+#   from mymodule import my_context_var
+#   CONTEXT_VARS_TO_PROPAGATE.append(my_context_var)
+CONTEXT_VARS_TO_PROPAGATE: list[contextvars.ContextVar] = [
+    _secret_fetcher_context,
+]
+
+# Default timeout for request/response cycles
+TIMEOUT_SECONDS = 60.0
```

Comment:
Definitely going to want to make this configurable. Maybe even on a per-user basis. I wonder how many bitcoin you can mine in 60 seconds 🤔

Reply:
Good call, I'll make it tunable. Remember, this is not real user-code, so no mining could occur.